### PR TITLE
feat(jwk): cap RSA key size to 4096 bits in audience registration

### PR DESCRIPTION
**Pull request by crucible-burnt**
## Summary

Enforce a maximum RSA key size of 4096 bits when creating or updating JWK audiences. This prevents potential abuse via oversized RSA keys that are cheap to generate but expensive to verify against.

## Changes

- Add `validateJWKKeySize()` helper in `x/jwk/types/messages_audience.go`
- Apply key size check in `ValidateBasic()` for both `MsgCreateAudience` and `MsgUpdateAudience`
- RSA keys > 4096 bits are rejected at message validation layer
- ECDSA/Ed25519 keys are inherently bounded and pass through
- Add comprehensive tests in `x/jwk/types/key_size_test.go`

## Linear

- ENG-1478

## Testing

- Unit tests for valid 2048-bit, valid 4096-bit, and rejected 8192-bit RSA keys
- Tests cover both CreateAudience and UpdateAudience paths

### CODE REVIEW

**Review by crucible-burnt — COMMENTED** [review]
## 🔍 Crucible Security Review

### Summary
Solid defensive hardening PR. Caps RSA JWK key size at 4096 bits in `ValidateBasic()` for both `MsgCreateAudience` and `MsgUpdateAudience`, preventing DoS via oversized keys that are cheap to generate but expensive to verify.

### Security Assessment
- **Risk Level**: Clean ✅
- Validation placement in `ValidateBasic()` is correct — rejects oversized keys before they hit state or signature verification
- Both `MsgCreateAudience` and `MsgUpdateAudience` are covered
- Handles both `*rsa.PublicKey` and `*rsa.PrivateKey` raw types (private keys can appear in JWK format)
- ECDSA and Ed25519 correctly identified as inherently bounded by curve/algorithm

**Minor observation (not blocking):**
- `messages_audience.go:47` — The `default` case silently allows unknown key types through. This is a reasonable design choice (fail-open for future key types), but worth documenting the intent. If a new key type is added to `lestrrat-go/jwx` with variable sizing (e.g., post-quantum), it would bypass this check. Consider a comment noting this is intentional.

### Immunefi Pattern Check
- ✅ No matches against known vulnerability classes
- This PR is *proactive defense* — no existing Immunefi reports target RSA key size abuse, but it closes a potential DoS vector before it's reported
- Related context: JWK module has been a frequent target (JWT replay, TimeOffset). Hardening here is welcome

### False Report Risk
- Low. The 4096-bit cap is well-documented in the constant comment and aligns with industry standards
- Test coverage clearly demonstrates the boundary (4096 accepted, 8192 rejected)

### Code Quality Notes
- Good test coverage: boundary cases at 2048, 4096, and 8192 bits
- Clean error messages with actual vs max bit lengths
- `generateOversizedRSAJWK` helper is well-structured

### Status
Clean from a security perspective. No blocking issues. The minor note about the `default` case is informational only.

File: x/jwk/types/messages_audience.go [hunk 1]
```diff
@@ -12,6 +15,45 @@ import (
...
+	MaxRSAKeyBits = 4096
+)
+
+// ValidateJWKKeySize checks that the raw key material does not exceed
+// the allowed maximum sizes. This prevents denial-of-service attacks
+// via oversized keys that are cheap to generate but expensive to verify against.
+//
+// This function is exported and reusable to allow key size validation
+// in multiple contexts including JWT verification and genesis validation.
+func ValidateJWKKeySize(key jwk.Key) error {
```
[the remark this answers is not in the document]
Comment (2xburnt) [reply x/jwk/types/messages_audience.go:34]: The exported naming is intentional for reuse across keeper and ante decorator validation paths.

File: x/jwk/types/key_size_test.go [hunk 2]
```diff
@@ -0,0 +1,73 @@
...
+		require.NoError(t, err)
+	})
+
+	t.Run("oversized RSA key rejected in CreateAudience", func(t *testing.T) {
+		oversizedKey := generateOversizedRSAJWK(t, types.MaxRSAKeyBits*2)
+		msg := types.NewMsgCreateAudience(admin, "https://test.example.com", oversizedKey)
+		err := msg.ValidateBasic()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "exceeds maximum allowed")
+	})
```
[the remark this answers is not in the document]
Comment (2xburnt) [reply x/jwk/types/key_size_test.go:32]: Acknowledged. Using errors.Is/errors.As would be cleaner but requires wrapping with sentinel errors which is a larger refactor. Substring matching is acceptable for test assertions that verify error messages from the same package.

File: x/jwk/types/key_size_test.go [hunk 3]
```diff
@@ -0,0 +1,73 @@
...
+
+	t.Run("oversized RSA key rejected in UpdateAudience", func(t *testing.T) {
+		oversizedKey := generateOversizedRSAJWK(t, types.MaxRSAKeyBits*2)
+		msg := types.NewMsgUpdateAudience(admin, admin, "https://test.example.com", "", oversizedKey)
+		err := msg.ValidateBasic()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "exceeds maximum allowed")
+	})
```
[the remark this answers is not in the document]
Comment (2xburnt) [reply x/jwk/types/key_size_test.go:40]: Acknowledged. Using errors.Is/errors.As would be cleaner but requires wrapping with sentinel errors which is a larger refactor. Substring matching is acceptable for test assertions that verify error messages from the same package.

File: x/jwk/types/key_size_test.go [hunk 4]
```diff
@@ -0,0 +1,73 @@
...
+		err := msg.ValidateBasic()
+		require.NoError(t, err)
+	})
+}
+
+// generateOversizedRSAJWK creates a JWK JSON string with a mock RSA key of the
+// given bit length. Instead of calling rsa.GenerateKey (which is slow for large
+// keys and flaky in CI), we construct an rsa.PublicKey with a synthetic N of the
+// desired bit length. The validation logic only inspects the modulus size, so a
+// cryptographically valid key is not required.
```
[the remark this answers is not in the document]
Comment (2xburnt) [reply x/jwk/types/key_size_test.go:61]: Good catch on the comment. However this is a test helper — the synthetic key construction is sufficient to trigger the size validation being tested.

**Comment by 2xburnt** [comment]
Rebased to GPG-sign all commits. All commits now verified (G) via `git log --show-signature`.

**Review by 2xburnt — DISMISSED** [review]

**Review by 2xburnt — APPROVED** [review]
